flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/RequiredRedundantResource.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.resourcemanager.slotmanager;
+
+import org.apache.flink.api.common.resources.CPUResource;
+import org.apache.flink.configuration.MemorySize;
+import org.apache.flink.util.Preconditions;
+
+/** Immutable profile of the required redundant resources. */
+public final class RequiredRedundantResource {
+    private final CPUResource cpuResource;
+    private final MemorySize memorySize;
+
+    public RequiredRedundantResource(CPUResource cpuResource, MemorySize memorySize) {
+        this.cpuResource = Preconditions.checkNotNull(cpuResource);
+        this.memorySize = Preconditions.checkNotNull(memorySize);
```

Comment:
I'm not sure if considering only the total memory is enough. And I am a bit concerned that this may result in resources for certain tasks that need to be redeployed not being met, even if we ensure that the redundant total memory is sufficient.

Reply:
Good suggestion.

It makes more sense to consider subdivision resource from a semantic point of view.

Then I don't need to introduce a new RequiredRedundantResource to manage the use of redundant resources. Instead, using ResourceProfile is sufficient

Looking forward to your opinion as well @xintongsong